packages/template-retail-react-app/app/request-processor.js

```diff
@@ -9,9 +9,6 @@
 // 'request-processor.js' and update the processRequest function so that
 // it processes requests in whatever way your project requires.
 
-// Uncomment the following line for the example code to work.
-// import {QueryParameters} from 'pwa-kit-react-sdk/utils/ssr-request-processing'
```

Comment:
This `QueryParameters` class is a custom version of the javascript native API `URLSearchParams`. I don't see a reason why we have to use it?

Reply:
Looking at the code in the SDK, the reason that we are using this custom `QueryParameters` class is because the order of queries is important (`?a=1&b=2` might be a hit, but `?b=1&a=2` might not). The `QueryParameters` is here to maintain order when you are manipulating the search string.

Although not using that class doesn't effect this particular situation, the removal of the example boilerplate code might mean that partners that do decide to make changes to the request processor will do so incorrectly not knowing about this restriction.

Reply:
`URLSearchParams` also preserves the order; the key differences seem to be how `null` is handled and how spaces are encoded.